### ISSUE TITLE
Workflow: Add example for workflow ID reuse policy

### ISSUE THREAD

**Issue by kaibocai** [dapr/docs#3909]
**What content needs to be created or modified?**

Reference to https://github.com/dapr/dapr/issues/7101 and https://github.com/dapr/dapr/pull/7308, workflow ID reuse policy support is ready. 

**Describe the solution you'd like**

Add examples for explaining workflow ID reuse policy

**Where should the new material be placed?**

Examples should be added somewhere here https://github.com/kaibocai/docs/tree/7e79a7f0bb725da707a3e2a713925b74ac74b877/daprdocs/content/en/developing-applications/building-blocks/workflow

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**

https://github.com/dapr/dapr/issues/7101
https://github.com/dapr/dapr/pull/7308

**Additional context**

**Comment by kaibocai** [comment]
I just recall that the SDK updates for reuse ID were paused when we worked on the .NET SDK updates for ID reuse support, the discussion about refactoring the design https://github.com/dapr/dapr/issues/6434 was brought up by the .NET SDK repo maintainer and the SDK updates are paused. Currently, customers cannot use this feature as SDKs don't have support for it. 
I suggest pushing this doc update to 1.14 after we finish refactoring the design and adding support for SDKs. 

cc @cgillum

**Comment by cgillum** [comment]
@kaibocai I don't think it makes sense to pause all SDK work because of a small change we want to make to the `SkipIfExists` policy. IIRC, the proposed change only impacts runtime and API behavior, but not the SDK behavior. I think we should do the work to complete the SDKs so that developers can start using them and getting value from them in the 1.13 timeframe (which was part of our published 1.13 roadmap, right?). We can make adjustments as necessary in 1.14 when we can make a corresponding runtime change but I don't see any reason to pause SDK work for something as small as this.

**Comment by kaibocai** [comment]
Thank you, @cgillum. I'll proceed with the SDK development. In fact, Java already had a PR for review concerning this feature. I plan to collaborate with another SDK owner to incorporate support for this feature. However, given our current timeline, it's possible that this update may not be included in the 1.13 release.

**Comment by msfussell** [comment]
@cgillum @kaibocai - What was the conclusion on this issue?

**Comment by msfussell** [comment]
@cgillum @kaibocai - Can one of you give an update on this. Is my understanding correct
-This feature was added in v1.13 and is not currently document (hence the PR)
- This needs to be added to all WOrkflow API in the SDKs (hence the example in the PR use DurableTaskFramework which is wrong)
- Do any of the SDKs plan to have this in v1.14 timeframe?

**Comment by cgillum** [comment]
My understanding at this point is that the feature was added, but only a subset of Durable Task SDKs support it. Right now, it might only be the Go SDK.

I took a quick look, and it appears that some PRs were created for .NET and Java, but ended up getting stalled for one reason or another:

* https://github.com/microsoft/durabletask-java/pull/188
* https://github.com/microsoft/durabletask-dotnet/pull/258

I'm not aware of whether any work was started for Python and JavaScript, nor am I aware of any timelines.

IIRC, 1.14 will be closing soon, so 1.15 might be more realistic for having this work done. However, work/discussions may be needed to get the appropriate teams/developers aligned on prioritizing and completing it.

**Comment by hhunter-ms** [comment]
This isn't supported in any Dapr SDKs yet - pushing to 1.15.

**Comment by hhunter-ms** [comment]
@kaibocai @cgillum should this be bumped to 1.16?

**Comment by jpeffer** [comment]
Is there a reason this was removed from the 1.17 tracking board? Has SDK support been prioritized yet?
